### PR TITLE
fix: fixed support for Spring EnableMethodSecurity (#15059)

### LINKED ISSUE #15059: Application doesn't start when enabling @EnableMethodSecurity with vaadin 24.0.0.alpha2
### Description of the bug

Application doesn't start when enabling @EnableMethodSecurity with vaadin 24.0.0.alpha2.

The problem is in VaadinWebSecurity where the servletContext is fetched. When enabling global method security the SPEL `@Value("#{servletContext.contextPath}")` can't be evaluated because there is no bean named servletContext available.

To reproduce create a new project with vaadin starter with vaadin 24 and security enabled. Then add the following class 

```
@Configuration
@EnableMethodSecurity(jsr250Enabled = true)
public class GlobalSecurityConfiguration {
    @Bean
    MethodSecurityExpressionHandler methodSecurityExpressionHandler() {
        return new DefaultMethodSecurityExpressionHandler();
    }
}
```

### Expected behavior

The application should start

### Minimal reproducible example

See [https://github.com/aspan/my-app](https://github.com/aspan/my-app)

Start with `mvn spring-boot:run`

### Versions

- Vaadin / Flow version: 24.0.0.alpha2
- Java version: 17.0.5
- OS version: N/A
- Browser version (if applicable):
- Application Server (if applicable):
- IDE (if applicable):

### PR DESCRIPTION
**Pull request by MarcinVaadin**
Vaadin clone of https://github.com/vaadin/flow/pull/15671 plus tests

Fixes #15059

Co-authored-by @aspan

### CODE REVIEW

**Comment by MarcinVaadin** [comment]
@aspan - I've copied your solution and added test. It works as expected, thanks for contributing 👍

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinAwareSecurityContextHolderStrategyConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Registers {@link SecurityContextHolderStrategy} bean.
+     * <p>
+     * Beans of this type will automatically be used by
+     * {@link org.springframework.security.config.annotation.method.configuration.GlobalMethodSecurityConfiguration}
+     * to configure the current {@link SecurityContextHolderStrategy}.
+     */
+    @Bean(name = "VaadinSecurityContextHolderStrategy")
+    public SecurityContextHolderStrategy securityContextHolderStrategy() {
+        VaadinAwareSecurityContextHolderStrategy vaadinAwareSecurityContextHolderStrategy = new VaadinAwareSecurityContextHolderStrategy();
```
[the remark this answers is not in the document]
Comment (MarcinVaadin) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinAwareSecurityContextHolderStrategyConfiguration.java:38]: @sonatype-lift ignoreall

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinAwareSecurityContextHolderStrategyConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        // Use a security context holder that can find the context from Vaadin
+        // specific classes
+        SecurityContextHolder.setContextHolderStrategy(
+                vaadinAwareSecurityContextHolderStrategy);
+        return vaadinAwareSecurityContextHolderStrategy;
+    }
+
+}
```
Comment (mcollovati) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinAwareSecurityContextHolderStrategyConfiguration.java:46]: new line missing
```suggestion
}

```

**Review by mcollovati — CHANGES_REQUESTED** [review]
Just a new line missing, other than that LGTM

**Review by mcollovati — APPROVED** [review]

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinAwareSecurityContextHolderStrategyConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    /**
+     * Registers {@link SecurityContextHolderStrategy} bean.
+     * <p>
+     * Beans of this type will automatically be used by
+     * {@link org.springframework.security.config.annotation.method.configuration.GlobalMethodSecurityConfiguration}
+     * to configure the current {@link SecurityContextHolderStrategy}.
+     */
+    @Bean(name = "VaadinSecurityContextHolderStrategy")
+    public SecurityContextHolderStrategy securityContextHolderStrategy() {
```
Comment (knoobie) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinAwareSecurityContextHolderStrategyConfiguration.java:37]: Should this return `VaadinAwareSecurityContextHolderStrategy` instead? 

> When declaring a @Bean method, provide as much type information as possible in the method’s return type. For example, if your bean’s concrete class implements an interface the bean method’s return type should be the concrete class and not the interface. Providing as much type information as possible in @Bean methods is particularly important when using bean conditions as their evaluation can only rely upon to type information that’s available in the method signature.

Spring Docs: https://docs.spring.io/spring-boot/docs/2.4.2/reference/html/spring-boot-features.html#boot-features-bean-conditions 

Also, should this Bean also add `@ ConditionalOnMissingBean` to make sure it's easily override able by users if needed? Without using Primary hacks

Comment (MarcinVaadin) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinAwareSecurityContextHolderStrategyConfiguration.java:37]: I agree for both, thanks 👍

**Review by mcollovati — APPROVED** [review]

**Pull request merged**